### PR TITLE
[test-only]PodSecurity: make integration tests run sparsely

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind flake

#### What this PR does / why we need it:

Makes the PodSecurity integration tests cheaper to run by only testing levels that have policy or fixture changes.

xref #103512 

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/pod-security-admission/test/run.go [hunk 1]
```diff
@@ -82,22 +83,64 @@ func checksForLevelAndVersion(checks []policy.Check, level api.Level, version ap
...
+		// include the release under development (1.22 at time of writing).
+		// this can be incremented to the current version whenever is convenient.
+		api.MajorMinorVersion(1, 22),
+	}
+	for _, version := range alwaysIncludeVersions {
+		if seenVersions[version] {
+			continue
 		}
+		seenVersions[version] = true
+		versions = append(versions, version)
```
Comment (tallclair) [inline staging/src/k8s.io/pod-security-admission/test/run.go:139]: nit: I'd prefer to just construct `versions` at the end by iterating over `seenVersions`

File: staging/src/k8s.io/pod-security-admission/test/run.go [hunk 2]
```diff
@@ -82,22 +83,64 @@ func checksForLevelAndVersion(checks []policy.Check, level api.Level, version ap
...
-		if lastCheckVersion.Minor() > maxTestMinor {
-			maxTestMinor = lastCheckVersion.Minor()
+	}
+
+	alwaysIncludeVersions := []api.Version{
+		// include the oldest version by default
+		api.MajorMinorVersion(1, 0),
+		// include the release under development (1.22 at time of writing).
+		// this can be incremented to the current version whenever is convenient.
+		api.MajorMinorVersion(1, 22),
```
Comment (tallclair) [inline staging/src/k8s.io/pod-security-admission/test/run.go:132]: I'd rather just use `latest` here, but that would require a bit of work to make `getFixtures` and `getMinimalValidPod` work with latest. Maybe just put a TODO for now?

File: staging/src/k8s.io/pod-security-admission/test/run.go [hunk 3]
```diff
@@ -140,15 +183,12 @@ func Run(t *testing.T, opts Options) {
...
 	}
-	maxMinor, err := maxMinorVersionToTest(opts.Checks)
-	if err != nil {
-		t.Fatalf("invalid checks: %v", err)
-	}
 
-	for _, level := range []api.Level{api.LevelBaseline, api.LevelRestricted} {
-		for minor := 0; minor <= maxMinor; minor++ {
-			version := api.MajorMinorVersion(1, minor)
+	versionsToTest := computeVersionsToTest(t, opts.Checks)
```
Comment (tallclair) [inline staging/src/k8s.io/pod-security-admission/test/run.go:187]: optional: consider exposing this approach in options (e.g. `opts.RunAllVersions`)

File: staging/src/k8s.io/pod-security-admission/test/run.go [hunk 4]
```diff
@@ -140,15 +183,12 @@ func Run(t *testing.T, opts Options) {
...
 
+	for _, level := range []api.Level{api.LevelBaseline, api.LevelRestricted} {
```
[the remark this answers is not in the document]
Comment (tallclair) [reply staging/src/k8s.io/pod-security-admission/test/run.go:189]: optional: We don't really need to run the baseline tests for a version that only has changes in the restricted tests, but that might be over optimizing at this point.

**Review by tallclair — APPROVED** [review]
approach looks good, just some minor nits.

File: staging/src/k8s.io/pod-security-admission/test/run.go (hunk 3 above)
Comment (liggitt) [reply staging/src/k8s.io/pod-security-admission/test/run.go:187]: added a TODO to consider that

File: staging/src/k8s.io/pod-security-admission/test/run.go (hunk 2 above)
Comment (liggitt) [reply staging/src/k8s.io/pod-security-admission/test/run.go:124]: added a TODO

File: staging/src/k8s.io/pod-security-admission/test/run.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/pod-security-admission/test/run.go]: done

File: staging/src/k8s.io/pod-security-admission/test/run.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (liggitt) [reply staging/src/k8s.io/pod-security-admission/test/run.go:184]: not yet. this line didn't change, not sure why github is showing it as a diff

**Comment by liggitt** [comment]
green pull-kubernetes-integration run... /shrug

**Pull request merged**